src/Data/Int.purs

```diff
@@ -77,6 +79,21 @@ foreign import toNumber :: Int -> Number
 fromString :: String -> Maybe Int
 fromString = fromStringAs (Radix 10)
 
+-- | A type for describing whether an integer is even or odd.
+data Parity = Even | Odd
+
+derive instance eqParity :: Eq Parity
+derive instance ordParity :: Ord Parity
```

Comment:
Can you give me a link?

I think there can only be one ring with 2 elements. Suppose we have an arbitrary ring with 2 elements. First, 0 and 1 must be distinct, as if they were not, the product of 0 (=1) with the other element would have to be equal to 0 (since anything times 0 equals 0) but also equal to itself (as 1 is the multiplicative identity). So 0 and 1 cannot be the same, so we can call the two elements of this ring 0 and 1. Then,

* 0 + 0 = 0, and 0 + 1 = 1 + 0 = 1 (as 0 is the identity of +)
* 1 + 1 = 0 (since + forms a group)
* 1 * 1 = 1 (as 1 is the identity of *)
* 0 * 0 = 1 * 0 = 0 * 1 = 0 (anything times 0 is 0)

This is precisely how both Z/2Z behaves, and also precisely how booleans with `xor` as addition and `&&` as multiplication behave. It's also commutative, and every nonzero element (i.e. just 1) has a multiplicative inverse (itself) so it's a field.